bite-size-notes/week-5-e-SCvx-and-GuSTO-SCP.md

```diff
@@ -0,0 +1,127 @@
+Most of the convexification of (1) is done through the first-order approximation at the reference trajectory by computing the following jacobian matrices:
+
+|Equation | #|
+|-----|----|
+|$$A(t) \triangleq \nabla_{x}f(t, \bar{x}(t), \bar{u}(t), \bar{p}),$$| (5a)|
+|$$B(t) \triangleq \nabla_{u}f(t, \bar{x}(t), \bar{u}(t), \bar{p}),$$| (5b)|
+|$$F(t) \triangleq \nabla_{p}f(t, \bar{x}(t), \bar{u}(t), \bar{p}),$$| (5c)|
+|$$r(t) = f(t, \bar{x}(t), \bar{u}(t), \bar{p}) - A\bar{x}(t) -B\bar{u}(t) - F\bar{p}$$| (5d)|
+|$$C(t) \triangleq \nabla_{x}s(t, \bar{x}(t), \bar{u}(t), \bar{p}),$$| (5e)|
+|$$D(t) \triangleq \nabla_{u}s(t, \bar{x}(t), \bar{u}(t), \bar{p}),$$| (5f)|
+|$$G(t) \triangleq \nabla_{p}s(t, \bar{x}(t), \bar{u}(t), \bar{p}),$$| (5g)|
+|$$r'(t) = s(t, \bar{x}(t), \bar{u}(t), \bar{p}) - C\bar{x}(t) -D\bar{u}(t) - G\bar{p}$$| (5h)|
+|$$H_{0} \triangleq \nabla_{x}g_{ic}(\bar{x}(0), \bar{p}),$$| (5i)|
+|$$K_{0} \triangleq \nabla_{p}g_{ic}(\bar{x}(0), \bar{p}),$$| (5j)|
+|$$l_{0} \triangleq g_{ic}(\bar{x}(0), \bar{p}) - H_{0}\bar{x}(0) - K_{0}\bar{p},$$| (5k)|
+|$$H_{f} \triangleq \nabla_{x}g_{tc}(\bar{x}(1), \bar{p}),$$| (5l)|
+|$$K_{f} \triangleq \nabla_{p}g_{tc}(\bar{x}(1), \bar{p}),$$| (5m)|
+|$$l_{f} \triangleq g_{tc}(\bar{x}(1), \bar{p}) - H_{f}\bar{x}(1) - K_{f}\bar{p}.$$| (5n)|
+
+and the linearized (1) becomes:
+
+|Equation | #|
+|-----|----|
+|$$\min_{u,p} J(x, u, p),| (6a)|
+|$$s.t. \quad \dot{x}(t) = A(t)x(t) + B(t)u(t) + F(t)p(t) + r(t),$$| (6b)|
+|$$(x(t), p) \in \mathcal{X}(t),$$| (6c)|
+|$$(u(t), p) \in \mathcal{U}(t),$$| (6d)|
+|$$C(t)x(t) + D(t)u(t) + G(t)p + r'(t) \leq 0,$$| (6e)|
+|$$H_{0}x(0) + K_{0}p + l_{0},$$| (6f)|
+|$$H_{f}x(1) + K_{f}p + l_{f} = 0.$$| (6g)|
+
+Notice that the cost is not convexified while everything else is convex in (6). The cost function will be convexified through SCvx and GuSTO algorithm to be discussed. The control trajectory $u(.)$ is an infinite-dimensional vector space of continuous-time functions and needs to be discretized through temporal discretization and direct colocation so it can be solved numerically.
+
+Linearization guarantees that constraints in (6) are convex but it introduces **Artificial Unboundedness** where the subproblem solution must be kept close to the linearization point and sometimes the convex cost can go to negative infinity.
+
+Artificial Unboundedness can be solved by adding the following trust region constraint:
+
+|Equation | #|
+|-----|----|
+|$$\delta x(t) = x(t) - \bar{x}(t),$$
+$$\delta u(t) = u(t) - \bar{u}(t),$$
+$$\delta p = p - \bar{p},$$
+$$\alpha_{x}\parallel \delta x(t) \parallel_{q} + \alpha_{u}\parallel \delta u(t) \parallel \leq \eta, for t \in [0,1],$$| (7)|
+for some choice of $q \in {1,2,2^{+}, \inf}$ and constants $\alpha_{x}, \alpha_{u}, \alpha_{p} \in {0,1}. Use $q = 2^{+} to denote the ability to impose the trust region as the quadratic two-norm squared. The trust region radius $\eta$ is a fixed scalar that is updated between SCP iterations.
```

Comment:
This part is not properly enclosed by $.
It should be the following:

for some choice of $q \in {1,2,2^{+}, \inf}$ and constants $\alpha_{x}, \alpha_{u}, \alpha_{p} \in {0,1}$. Use $q = 2^{+}$ to denote the ability to impose the trust region as the quadratic two-norm squared. The trust region radius $\eta$ is a fixed scalar that is updated between SCP iterations.

Reply:
Yup, I missed this and I did not double-check. It is fixed now!